libHVCAENx527App/src/CAENHVWrapper_sim.cpp

```diff
@@ -40,11 +43,14 @@ CAENHVLIB_API CAENHVRESULT  CAENHVDeinitSystem(const char *SystemName)
 CAENHVLIB_API CAENHVRESULT  CAENHVGetChName(const char *SystemName, ushort slot, 
  ushort ChNum, const ushort *ChList, char (*ChNameList)[MAX_CH_NAME])
 {
-    // ChNum should be 1
-	*(reinterpret_cast<void**>(ChNameList)) = malloc(MAX_CH_NAME * ChNum);
+    if (++g_call_out % g_error_on_call == 0) {
+	    return CAENHV_TIMEERR;
```

Comment:
It would be better for us to implement a backdoor for `error on next read` or `error forever` etc, rather than relying on erroring every nth request or erroring randomly. This way we could integrate it into the test framework too.